### ISSUE TITLE
Does anyone maintain the windows platform? What is the specific release time of the latest version?

### ISSUE THREAD

**Issue by qd1332543** [Project-OSRM/osrm-backend#6135]

**Comment by jcoupey** [comment]
> Does anyone maintain the windows platform

The fair answer here is probably "no". The dedicated [wiki page](https://github.com/Project-OSRM/osrm-backend/wiki/Windows-Compilation) is rather clear on the fact that the whole windows pipeline is pretty much experimental (not to mention the date of the last build and page modification).

**Comment by qd1332543** [comment]
Well, I have already read the instructions on the wiki page. you know, In my opinion, after so many years, the windows platform should have been supported. Anyway, thank you for your answer.

**Comment by danpat** [comment]
> you know, In my opinion, after so many years, the windows platform should have been supported

Doing this for an open source project requires someone with the time, resources, and interest to maintain.  Currently, this project has none of those.  Unfortunately, it takes work.

**Comment by danpat** [comment]
Hmm - I wonder how hard it would be to publish the artifacts from appveyor.  I have _no_ idea where the infrastructure behind `http://build.project-osrm.org/` lives.

**Comment by mjjbell** [comment]
Appveyor has a Github Releases integration, so it looks achievable. Auth is via personal access tokens though, so not ideal.
And it looks like it has trouble running all the [tests](https://github.com/Project-OSRM/osrm-backend/blob/master/appveyor.yml#L31).

Building in Github Actions might be a more maintainable approach.

**Comment by hoerup** [comment]
Well it seems like the GH actionsWin2019 platform includes VC++ and CMake so maybe it can be done this way

https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md

**Comment by porkedpie** [comment]
I was never successful in building locally, but the binaries (http://build.project-osrm.org/) work fine.

I would love to be able to get libosrm C++ working instead of using the slow HTTP API (ideally via Python) but despite finding https://pypi.org/project/pyosrm/ I could never get it to work since you need to compile locally.

In any case - while a bit slow, the pre-built binaries work fine.

**Comment by ssuluh** [comment]
I'm curious for those who are interested in windows platform what are your use cases ?  Windows desktop ? Running in Windows servers ? Are you using libosrm directly or using .net ?

**Comment by porkedpie** [comment]
For me, running on Desktop, using for various simulation applications.
Currently having to use a local osrm sever and calling the API via Python.
Even asynchronously this is pretty slow. Would be great to be able to call
the C code directly without the http overhead.

**Comment by jeanbaptisteb** [comment]
> I'm curious for those who are interested in windows platform what are your use cases ? Windows desktop ? Running in Windows servers ? Are you using libosrm directly or using .net ?

@ssuluh : my use case is running on windows desktop, calling the HTTP API via Python or R. The binaries from http://build.project-osrm.org/ work fine, but as far as I understand, they are not based on the latest version of OSRM. While the binaries work like a charm with this [Python package](https://pypi.org/project/osrm-py/), I ran into a couple of problems when trying to use [this R package](https://cran.r-project.org/web/packages/osrm) (which apparently requires the latest version of OSRM). That's why I'm looking for a way of running the latest version of OSRM on Windows. 

I can't really use Docker, relative to their terms of use. I'm thinking of running an OSRM server in some Linux virtual box running on my desktop (while sending HTTP requests from Windows to the OSRM virtual box server), but I still have to try it.

**Comment by jcoupey** [comment]
> I'm thinking of running an OSRM server in some Linux virtual box

IMHO probably your best bet if you want to keep things simple and easy.

**Comment by nilsnolde** [comment]
WSL2 (the integrated Linux system) might also work for Windows. I only ever used it to run docker, never tried building (and running) software directly.

**Comment by porkedpie** [comment]
If your only reason to update is to be able to use R package osrm, then building a version in Linux or a container seems like a harder option than either just using curl in R to call directly, or working in Python. If you have a big body of R code using osrm package, then just use an older version of that? It worked fine in recent history (~1 year ago), so probably the previous version is fine.

**Comment by jeanbaptisteb** [comment]
> If your only reason to update is to be able to use R package osrm, then building a version in Linux or a container seems like a harder option than either just using curl in R to call directly, or working in Python. If you have a big body of R code using osrm package, then just use an older version of that? It worked fine in recent history (~1 year ago), so probably the previous version is fine.

Thanks, it's a solution I'll consider for the project I'm working on, but I'm a bit wary that it may not be a workable solution for other future projects. 

The [R package I'm referring to ](https://cran.r-project.org/web/packages/osrm/index.html) has some features not available in Python, as far as I know (i.e. [isochrone maps](https://rdrr.io/cran/osrm/man/osrmIsochrone.html)). To be honest, I'm not very keen on writing code based on an old version of the R package and on an old version of OSRM. I'm a little scared that at some point it could become some kind of dependency hell for complex projects, or to run into other bugs that have been corrected in later versions of OSRM or of the R package. I'd prefer to start on a "sound" basis to limit these risks. So anyway, I'll try the virtual box or WSL solutions that others suggested, and if it doesn't work I'll fall back on your suggestion. 

Thank you all for the feedback. If I have it working, I'll try to outline some instructions for other people on windows.

**Comment by nilsnolde** [comment]
FWIW, we still want to create a proper python package for OSRM with batteries included (ie the OSRM library), preferably with the build pipelines available via python API, so there’s no need for separate OSRM exes, it would all be on PyPI for Win, Mac & Linux. But.. time.. I did it for [Valhalla already](https://github.com/gis-ops/pyvalhalla), so it _should_ be a quick thing to do.

**Comment by jeanbaptisteb** [comment]
> Thank you all for the feedback. If I have it working, I'll try to outline some instructions for other people on windows.

Just as a short follow-up for people who'd want to use OSRM on Windows (desktop) without waiting: I gave up on trying to use WSL2 for that (encountered too many issues), and eventually I had it work with an Ubuntu guest (22.04.1 LTS) in a virtualbox. I simply followed the instructions for installing OSRM on Ubuntu, and configured the virtualbox network as NAT. Then, in the virtualbox network advanced settings, I set the port forwarding options to 127.0.0.1:5000 for the host and 0.0.0.0:5000 for the guest. Now, I can send requests from my Windows desktop (Windows 10) to the OSRM server hosted on the Ubuntu virtualbox. Requests have to be sent to http://127.0.0.1:5000 or http://localhost:5000.
